test/GPv2Settlement.test.ts

```diff
@@ -783,47 +783,71 @@ describe("GPv2Settlement", () => {
   describe("executeInteractions", () => {
     it("executes valid interactions", async () => {
       const EventEmitter = await ethers.getContractFactory("EventEmitter");
-      const contract1 = await EventEmitter.deploy();
-      const contract2 = await EventEmitter.deploy();
-      const contract3 = await EventEmitter.deploy();
-      expect(contract1.address)
-        .not.to.equal(contract2.address)
-        .not.to.equal(contract3.address);
-      expect(contract2.address).not.to.equal(contract3.address);
+      const interactionParameters = [
+        {
+          target: await EventEmitter.deploy(),
+          value: ethers.utils.parseEther("0.42"),
+          number: 1,
+        },
+        {
+          target: await EventEmitter.deploy(),
+          value: ethers.utils.parseEther("0.1337"),
+          number: 1,
+        },
+        {
+          target: await EventEmitter.deploy(),
+          value: ethers.constants.Zero,
+          number: 1,
+        },
+      ];
```

Comment:
Either this:
```suggestion
      const interactionParameters = [
        {
          target: await EventEmitter.deploy(),
          value: ethers.utils.parseEther("0.42"),
          number: 1,
        },
        {
          target: await EventEmitter.deploy(),
          value: ethers.utils.parseEther("0.1337"),
          number: 2,
        },
        {
          target: await EventEmitter.deploy(),
          value: ethers.constants.Zero,
          number: 3,
        },
      ];
```
or, maybe better, inject the number directly in the for loops?

Reply:
:see_no_evil:

Reply:
> maybe better, inject the number directly in the for loops?

What I dislike about that is the "implicit" nature of the test parameters. In this case, since we are trying to verify that the value in `number` round-trips, I prefer having it very explicit.